### PR TITLE
Oxo game and Tests in oxoTest.js by Valerii Fedorenko

### PR DESCRIPTION
**Pull request by valeraf23**

### CODE REVIEW

File: contracts/OxoFactory.sol [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                   array[i] =  uint(keccak256("empty"));
+		 someGame.board =array;
+		 uint id = games.push(someGame) - 1;
+		 someGame.money = msg.value;
+		 gameById[id] = someGame;
+		 userByAddr[msg.sender].gameId = id;
+		 NewGameEvent("New Game was created", id );
+    }
+
+    function joinToGameById(uint _id) payable public onlyKnownUser onlyFinishGame returns(bool){
```
Comment (Purka) [inline contracts/OxoFactory.sol:59]: what will happen if I try to join to game that wasn't created? Do you have a test for this case?

Comment (valeraf23) [reply contracts/OxoFactory.sol:59]: yeap. You are right.
I have added verification and several tests for it. Please look again.

File: contracts/OXO.sol [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	 function doStep(uint _field) stillCanPlay() onlyWaitThisPlayer() onlyEmptyCell(_field) public{
+	   Game memory game = gameById[getGameId()];
+		game.board[_field] = (msg.sender == game.playerOne)? uint(keccak256("x")) : uint(keccak256("o"));
+		State _state = game.state;
+		game.countOfMove ++ ;
+		if(game.countOfMove >= 5){
+		    _state = checkWinner(game, _field);
+			if(_state != State.WinnerPlayerOne && _state != State.WinnerPlayerTwo ){
+		    	_state = checkDraw(game);
```
Comment (Purka) [inline contracts/OXO.sol]: Does it make sense to check for a draw when countOfMove < 9?

File: contracts/OXO.sol [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    		_state =  stateForNextPlayer(game);
+		}
+		StepEvent(msg.sender, _field, _state);
+		game.state = _state;
+		game.status = stateToString(_state);
+		game.next = (msg.sender == game.playerOne)? game.playerTwo : game.playerOne;
+		gameById[getGameId()] = game;
+    }
+
+    function checkDraw(Game _game) private returns(State){
```
Comment (Purka) [inline contracts/OXO.sol:31]: If you use the same number of indents through the contract the universe will be benevolent with you

File: contracts/OXO.sol (hunk 2 above)
Comment (valeraf23) [reply contracts/OXO.sol]: inside 'checkDraw(game)' was the condition that checking execute when the count of steps was equal 9.
Nonetheless, you are right, as result, I have changed logic checking 'Draw', and add the condition that checks 'Draw' if a count of move more or equal 8

File: contracts/OXO.sol (hunk 3 above)
Comment (valeraf23) [reply contracts/OXO.sol]: :-)

File: contracts/OXO.sol (hunk 2 above)
Comment (valeraf23) [reply contracts/OXO.sol]: Please  , Look again

File: contracts/OXO.sol (hunk 3 above)
Comment (valeraf23) [reply contracts/OXO.sol:31]: Strange
I did format code http://take.ms/5KoVe but when I look on GitHub I still have not right  number of indents

**Review by Purka — APPROVED** [review]

File: contracts/OXO.sol (hunk 3 above)
Comment (lastperson) [reply contracts/OXO.sol:31]: @valeraf23 tabs and spaces mixed together.

**Pull request closed without merging**